### PR TITLE
[Torch-MLIR] add support for clip model

### PR DESCRIPTION
**Pull request by zzpmiracle**

### CODE REVIEW

File: pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp [hunk 1]
```diff
@@ -1398,7 +1398,7 @@ class ShapePropagator : public PropertyPropBase {
             if (maybe_dtype_option && !maybe_dtype_option->isNone()) {
               return {ret->withScalarType(maybe_dtype_option->toScalarType())};
             } else {
-              return {ret};
+              return {};
```
Comment (tanyokwok) [inline pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp:1401]: Why not return the input type as it's?

**Comment by tanyokwok** [comment]
mark related https://github.com/alibaba/BladeDISC/issues/904

File: pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp (hunk 1 above)
Comment (zzpmiracle) [reply pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp]: OK, return null to skip some inputs like prim::dtype. I have dealt with prim::dtype now.

File: pytorch_blade/tests/disc/ops/test_factory_like.py [hunk 2]
```diff
@@ -49,7 +49,22 @@ def fulls_dtype(d0: int, d1: int, val: int, device: torch.device):
...
+        def empty(d0: int, d1: int, val: int, device: torch.device):
+            return torch.empty([d0, d1], device=device)
+
+        @torch.jit.script
+        def empty_dtype(d0: int, d1: int, val: int, device: torch.device):
+            return torch.empty([d0, d1],
+                              dtype=torch.float32, device=device)
 
+        # results are not numeric checkable
+        self._test_factory_like(empty, 2, 2, 3, rtol=1e+6, atol=1e+6)
```
Comment (tanyokwok) [inline pytorch_blade/tests/disc/ops/test_factory_like.py:66]: YYDS

File: pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp [hunk 3]
```diff
@@ -104,6 +104,27 @@ c10::Device getDeviceFromValue(Value* value) {
   return device;
 }
 
+c10::optional<at::ScalarType> getScalarTypeFromValue(Value* value) {
+  c10::optional<at::ScalarType> scalarType;
+  if (value == nullptr)
+    return scalarType;
+  if (value->node()->kind() == prim::dtype) {
```
Comment (tanyokwok) [inline pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp:111]: Nit: add a shape inference test for `prim::dtype`

Comment (zzpmiracle) [reply pytorch_blade/pytorch_blade/compiler/jit/torch/shape_analysis.cpp:111]: OK，both torchscript and e2e tested

**Review by tanyokwok — APPROVED** [review]
LGTM

**Pull request merged**